common/sha2.h

```diff
@@ -1,7 +1,27 @@
 #ifndef SHA2_H
 #define SHA2_H
 
-int sha384(uint8_t *out, const uint8_t *in, size_t inlen);
-int sha512(uint8_t *out, const uint8_t *in, size_t inlen);
+#include <stddef.h>
+#include <stdint.h>
+
+void sha224_inc_init(uint8_t *state);
+void sha224_inc_blocks(uint8_t *state, const uint8_t *in, size_t inblocks);
```

Comment:
these blocks are 64 bytes? Is that documented somewhere?

Reply:
Good point - I've added a few lines of documentation.